doc/ui_component2_transition.md

```diff
@@ -0,0 +1,132 @@
+# UiComponent2 Transition Notes
+> A document to explain the new features and breakages when converting components from Component / UiComponent to 
+Component2 / UiComponent2.
+---
+* __[What's New in Component2](#whats-new-in-component2)__
+* __[Why a new base class?](#why-a-new-base-class)__
+* __[New Features, improvements, and breakages](#new-features-improvements-and-breakages)__
+    * [Small Perf Improvements](#small-perf-improvements)
+    * [JS-Backed Maps](#js-backed-maps)
+    * [Lifecycle Method Changes](#lifecycle-method-changes)
+    * [Other Method Changes](#other-method-changes)
+    * [PropTypes](#prop-validation-goes-through-proptypes)
+    * [ErrorBoundary Component Added](#errorboundary-component-added)
+    * [Other Breaking Changes](#other-breakages-we-made-while-we-had-the-opportunity)
+* __[Updating](#updating)__
+## What's New in Component2
+`Component` and `UiComponent` are being deprecated in favor of new versions of the API: `Component2` and `UiComponent2`.
+
+#### tl;dr
+
+- Pros: slightly faster, improved dev experience, easier to maintain, easier integration with JS libs
+- Cons: breaks a few advanced component APIs that will require conversion, some of which will be automated with codemod
+
+Migration can happen at any time, and is opt-in. We'll provide a [codemod](#updating) to handle the simpler cases and 
+run it as 
+part of cleanup, but it will take manual intervention to update some components.
+
+## Why a new base class?
+- It's not possible for us to automatically support both unsafe and safe lifecycle methods without a resolved AST in 
+the over_react builder, which would increase build times dramatically.
+- We needed some small breakages to support
+  - JS-backed maps
+  - New lifecycle methods
+  - Injection of Dart->JS interop pieces into component instance (initComponentInternal) needed to change (this was 
+ publicly accessible, but is not done anywhere)
+  - componentDidUpdate, which now takes an additional `snapshot` argument
+  - Assignment of `props`/`state` to arbitrary Maps (was not advised, but possible)
```

Comment:
This line is confusing to me, did we make it possible and it IS not advised or was it possible (and not advised) and it is not possible anymore?

Reply:
I'm 99% sure we made it impossible to set the maps directly, and I updated the wording to reflect that, but leaving this unresolved for @greglittlefield-wf to confirm

Reply:
Now they can only be set to `JsBackedMap`s, but it's considered unsupported usage.